### ISSUE TITLE
Mapping Enums to custom type

### ISSUE THREAD

**Issue by jacktattersall1** [netglade/auto_mappr#80]
Hi, i currently have a project that uses drift db and mappr together.
Obviously, mapping types to drift's generated dataclass is trivial, however, i need to map types to drift's generated companion class.

Now, all fields of a companion class are types wrapped in a "Value"

So far i can achieve primitives mapping like so..

```
MapType<int, Value<int>>(fields: [
    Field('value', custom: Mappr.testInt),
    whenSourceIsNull: Value.absent()
  ]),
  
static int testInt(int int) => int;
```

however, when trying to map an Enum to a Value<Enum> i am running to issues with whatever i try.

```
MapType<DocFileType, Value<DocFileType>>(
      fields: [Field('value', custom: Mappr.testEnum)],
      whenSourceIsNull: Value.absent()),

static T testEnum<T>(T enu) => enu;
```

i get a `Failed to map DocFileType → Value<DocFileType> because target Value<DocFileType> is not an enum` error.
Now, according to the docs -
`The target enum can either be a superset of the source or has to define whenSourceIsNull which will be used for unknown enum values. If the target is not a superset of the source enum the generator will throw.`

i have also tried a generic `MapType<Enum, Value<Enum>>` with a custom function which appears to work, and generates
```
if ((sourceTypeOf == _typeOf<Enum>() || sourceTypeOf == _typeOf<Enum?>()) &&
        (targetTypeOf == _typeOf<Value<Enum>>() ||
            targetTypeOf == _typeOf<Value<Enum>?>())) {
      if (canReturnNull && model == null) {
        return null;
      }
      return (_map_Enum_To_Value$Enum((model as Enum?)) as TARGET);
    }
```

however, this doesnt pick up my enums whens mapping occurs.

Not sure what im doing wrong, or if this is even possible.

Any help appreciated.
Thanks

**Comment by jacktattersall1** [comment]
for clarity, here is companion class constructor

```
class DocumentTemplateEntitiesCompanion
    extends UpdateCompanion<DocumentTemplateEntity> {
  final Value<int?> localDatabaseId;
  final Value<int> id;
  final Value<String> name;
  final Value<DocFileType> docFileType;
  final Value<bool> isDeleted;
  const DocumentTemplateEntitiesCompanion({
    this.localDatabaseId = const Value.absent(),
    this.id = const Value.absent(),
    this.name = const Value.absent(),
    this.docFileType = const Value.absent(),
    this.isDeleted = const Value.absent(),
  });
```

and a Value type has 2 methods of construction

```
/// Create a (present) value by wrapping the [value] provided.
  const Value(T value)
      : _value = value,
        present = true;

  /// Create an absent value that will not be written into the database, the
  /// default value or null will be used instead.
  const Value.absent()
      : _value = null,
        present = false;
```

**Comment by tenhobi** [comment]
Hi, we already discussed this as "type converters" in #60. We still discussing the best way to allow such things at the moment. If you have some ideas and want to discuss them, either post them here or on our Discord. 😊

**Comment by tenhobi** [comment]
Type converters will be available in v2, now available using version 2.0.0-beta2
